cli/cli_test.go

```diff
@@ -430,37 +431,13 @@ func TestGetMenderDaemonPID(t *testing.T) {
 	assert.Error(t, updateCheck(cmdKill, cmdPID))
 }
 
-// Minimal init
-func TestInitDaemon(t *testing.T) {
-	// create directory for storing deployments logs
-	tempDir, _ := ioutil.TempDir("", "logs")
-	defer os.RemoveAll(tempDir)
-	app.DeploymentLogger = app.NewDeploymentLogManager(tempDir)
-	bootstrap := false
-	dualRootfs := installer.NewDualRootfsDevice(nil, nil, installer.DualRootfsDeviceConfig{})
-	d, err := initDaemon(&conf.MenderConfig{}, dualRootfs,
-		&runOptionsType{dataStore: tempDir, bootstrapForce: bootstrap})
-	require.Nil(t, err)
-	assert.NotNil(t, d)
-	// Test with failing init daemon
-	set := flag.NewFlagSet("", 0)
-	command := &cli.Command{Name: "daemon"}
-	ctx := cli.NewContext(nil, set, nil)
-	ctx.Command = command
-	runOpts := runOptionsType{
-		logOptions: logOptionsType{logLevel: "info"},
-	}
-	assert.Error(t, runOpts.handleCLIOptions(ctx))
-}
-
 // Tests that the client will boot with an error message in the case of an invalid server certificate.
 func TestInvalidServerCertificateBoot(t *testing.T) {
 	tdir, err := ioutil.TempDir("", "invalidcert-test")
 	require.Nil(t, err)
 
 	logBuf := bytes.NewBuffer(nil)
-	defer func(oldLog *log.Logger) { log.Log = oldLog }(log.Log) // Restore standard logger
-	log.Log = log.New()
+	var hook = logtest.NewGlobal()
 	log.SetLevel(log.WarnLevel)
```

Comment:
Do you need `defer hook.Reset()` here?

Reply:
Yes, you're right. Should probably be here. Although not stricly necessary, as the hook is only used in this function. I'm still undecided whether or not we should make some sort of utility test logger package, or maybe a test-global hook, or keep it like this. 

Thoughts?

Reply:
Yes, but it has global effects, doesn't it? So resetting it is important.

I don't think we should make a global test hook, because the test functions have different requirements. I think this particular call is better to do per test.

Reply:
Come to think of it, the effects should be local actually, as the hook can not be reach after it has passed out of scope. So the `defer hook.Reset()`, is actually unnecessary.

Reply:
It has [global effects](https://godoc.org/github.com/sirupsen/logrus/hooks/test#NewGlobal), namely changing the global logger. Otherwise it would not be very useful, how would the production code use the hook, otherwise?

Reply:
Yes, I thought this way first also. But it only adds the `hook` to the global logger. And the specific hook is only reachable (in this case) within the scope we attach it to the global logger. Which makes it local, since no one else is using this hook.

Does that make sense?

Reply:
Latest commit has the reset added in a defer statement

Reply:
Right, I see what you mean. After looking at the actual code, I don't think there is a way to remove the hook actually (I have observed this earlier as well), so after a long test run you may actually have a gazillion hooks attached to the logger that do nothing other than use memory. Not a problem with our code, it's just a design flaw of logrus.

Since you've added it, let's keep the defer for now anyway, but I see your point.